### PR TITLE
Move ReactElementValidator to __DEV__ block

### PR DESCRIPTION
**Pull request by ruiaraujo**
It saves some more bytes in production mode.

```
   raw     gz Compared to master @ fd589fc8dd5c8f987c6496b203f7f45f1f1fbb2e    
     =      = build/react-dom-server.js                                        
     =      = build/react-dom-server.min.js                                    
     =      = build/react-dom.js                                               
     =      = build/react-dom.min.js                                           
    +6     +1 build/react-with-addons.js                                       
 -1880   -660 build/react-with-addons.min.js                                   
    +6     -2 build/react.js                                                   
 -1875   -592 build/react.min.js       
```

### CODE REVIEW

**Comment by gaearon** [comment]
Can you post the sizes before and after? I believe they should be printed when you run the build.

**Comment by ruiaraujo** [comment]
@gaearon done above.

**Comment by jimfb** [comment]
Looks good to me.  Thanks!

**Comment by gaearon** [comment]
Wow, -600 bytes gzipped is great, thanks!

**Comment by ruiaraujo** [comment]
@gaearon low hanging fruit is the best fruit. ;)

**Comment by zpao** [comment]
Just coming around to this. This win is going to get lost internally at FB and in the future if we move to imports, both cases where dependencies are statically analyzed. It's not the only place we do this (IIRC) so it's totally cool to do, just wanted to fill in some backstory (and make sure other FB people understand).

**Comment by gaearon** [comment]
@zpao I totally broke my mind trying to parse your second sentence, can you rephrase please? 😄

**Comment by zpao** [comment]
I'm pretty sure we have another require() in dev blocks so it's ok to do here. I just wanted to reiterate that we only get the wins here and not at FB since requires are resolved statically.
